src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/Domain/LearnerCommitments.cs

```diff
@@ -22,8 +23,8 @@ public LearnerCommitments(IEnumerable<CommitmentEntity> commitments)
                 if (commitmentList.Count == 1)
                 {
                     // No 'versions'
-                    commitmentList[0].EffectiveStartDate = commitmentList[0].StartDate;
-                    commitmentList[0].EffectiveEndDate = commitmentList[0].EndDate;
+                    commitmentList[0].EffectiveStartDate = commitmentList[0].StartDate.LastDayOfMonth();
+                    commitmentList[0].EffectiveEndDate = commitmentList[0].EndDate.LastDayOfMonth().AddDays(-1);
```

Comment:
What happens if we have multiple versioned commitments and then an over lapping commitment? This only seems to get applied when there is just one unversioned commitment

Reply:
As discussed, but for future ref we determined that in all cases (due to the shape of the data - versioned commitments being dependant on the ILR that 'failed' them) that it would work in all cases